src/Microsoft.AspNet.OData/Routing/ODataPathRouteConstraint.cs

```diff
@@ -47,10 +47,32 @@ public virtual bool Match(HttpRequestMessage request, IHttpRoute route, string p
                 object oDataPathValue;
                 if (values.TryGetValue(ODataRouteConstants.ODataPath, out oDataPathValue))
                 {
+                    string oDataPath = oDataPathValue as string;
+                    Func<IServiceProvider> requestContainerFactory = () =>
+                    {
+                        // Delegate is reused. We need to ensure only one request container is created
+                        object requestContainer;
+                        if (request.Properties.TryGetValue(Extensions.HttpRequestMessageExtensions.RequestContainerKey, out requestContainer))
+                        {
+                            return (IServiceProvider)requestContainer;
+                        }
+
+                        return request.CreateRequestContainer(RouteName);
```

Comment:
>request.CreateRequestContainer(RouteName) [](start = 31, length = 41)

If there is a per-route request container, should it take precedence over one associated with the request? Couldn't CreateRequestContainer(RouteName) default to reqeust.Properties.TryGetValue(...) if there was no request container explicitly associated with the route?